### DISCUSSION TITLE
How can I control a feature of a dependency at configure time?

### DISCUSSION THREAD

**Discussion by kevin--** [microsoft/vcpkg#38158] [Q&A] [answered]
We have a dependency that has a feature that only works in some situations, and otherwise we need to not compile some aspects of our host project.

E.G.
dependency vcpkg.json in the registry:
```
{
  "name": "sick-library",
  "features": {
    "extra-auth": {
      "dependencies": [
        {
          "name": "jwt-cpp",
          "version>=": "0.6.0"
        }
      ]
    }
  }
}
```

then in the host vcpkg.json we just consume the default library `sick-library`.

At configure time however, if we wanted to add the feature how can we enable this?

```
option( DO_MORE_AUTH "..." ON )
if( DO_MORE_AUTH )
    list( APPEND VCPKG_MANIFEST_FEATURES "extra-auth" )
endif()
```
☝️ this does not work because `extra-auth` is not a feature of the host vcpkg.json.

Can we specify the feature of a dependency on the command line or in CMakeLists.txt somehow?

**Comment by Osyotr** [comment, marked as answer]
> Can we specify the feature of a dependency on the command line or in CMakeLists.txt somehow?

No.
> this does not work because extra-auth is not a feature of the host vcpkg.json.

It should be:
```json
{
  "name": "my-project",
  "dependencies": [
    "sick-library"
  ],
  "features": {
    "do-more-auth": {
      "dependencies": [
        {
          "name": "sick-library",
          "features": {
            "extra-auth"
          }
        }
      ]
    }
  }
}
```

**Reply by kevin--** [reply]
I see so we should roll the feature that we need into the feature of the host application. Thanks for clarifying
